pymbar/mbar.py

```diff
@@ -302,7 +303,33 @@ def __init__(self, u_kn, N_k, maximum_iterations=10000, relative_tolerance=1.0e-
                 # which might involve passing in different combinations of options, and passing out other strings.
                 solver['options']['verbose'] = self.verbose
 
-        self.f_k = mbar_solvers.solve_mbar_for_all_states(self.u_kn, self.N_k, self.f_k, solver_protocol)
+        np.random.seed(rseed)
+        self.rstate = np.random.get_state()
+
+        f_k_init = self.f_k.copy()  # we need to pass a copy so we don't overwrite the original 
+        self.f_k = mbar_solvers.solve_mbar_for_all_states(self.u_kn, self.N_k, f_k_init, solver_protocol)
+
+        self.nbootstraps = None
+        if nbootstraps != None:
+            self.nbootstraps = nbootstraps
+            #save the original data:
+            self.f_k_boots = np.zeros([self.nbootstraps,self.K])
+
+            allN = int(np.sum(self.N_k))
+            rinit = np.zeros(allN, int)
+            for b in range(self.nbootstraps):
+                for k in range(K): # randomize within the indices with the same K.
+                    # which of the indices are equal to K
+                    k_indices = np.where(self.x_kindices == k)[0]
+                    # pick new random ones of these K.
+                    new_kindices = k_indices[np.random.randint(int(N_k[k]), size=int(N_k[k]))]
+                    rinit[k_indices] = new_kindices
+
+                self.f_k_boots[b,:] = mbar_solvers.solve_mbar_for_all_states(self.u_kn[:,rinit], self.N_k, f_k_init, solver_protocol)
```

Comment:
@mrshirts  You mentioned that this PR already initializes this PR with the non-bootstrapped free energy estimate when solving the mbar equation for each bootstrapped free energy matrix, but I don't think it does. `f_k_init` here is a copy of self.f_k before the mbar equation is solved, meaning `f_k_init` is just zeros (see line 309). I can push a commit that moves line 309 to after 310, if you agree that that's the right fix here

Reply:
Hah, you are right, I was remembering what I did for pymbar4.  It is initializing from the previous bootstrap, which is still decent, but probably slightly worse than starting from the "real" value.   I will upload something in the next few min.